src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax;
+
+public class Main {
+    private static final FigureSupplier figureSupplier = new FigureSupplier();
+
+    public static void main(String[] args) {
+        Figure[] figures = new Figure[6];
+
+        for (int i = 0; i < figures.length; i++) {
+            figures[i] = figureSupplier.getRandomFigure();
+        }
+
+        for (int i = 0; i < figures.length / 2; i++) {
+            figures[i] = figureSupplier.getDefaultFigure();
+        }
```

Comment:
Overwriting the first half of the `figures` array with default figures is not necessary if you're already filling them with random figures. If you need to mix default and random figures, consider a different approach that does not overwrite the previously created figures.